### PR TITLE
`azurerm_storage_account` - Extend the support level of `(blob|queue|share)_properties` for `Storage` kind

### LINKED ISSUE #25422: `queue_properties` aren't supported for account kind "Storage" in sku tier "Standard"
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.5

### AzureRM Provider Version

3.97.1

### Affected Resource(s)/Data Source(s)

azurerm_storage_account

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "=3.97.1"
    }
  }
}

provider "azurerm" {
  features {}
}


resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_storage_account" "example" {
  name                     = "terraformissuestorage01"
  resource_group_name      = azurerm_resource_group.example.name
  location                 = azurerm_resource_group.example.location
  account_tier             = "Standard"
  account_kind             = "Storage"
  account_replication_type = "LRS"

  queue_properties {
    logging {
      delete                = true
      read                  = true
      version               = "1"
      write                 = true
      retention_policy_days = "10"
    }
  }

  tags = {
    environment = "staging"
  }
}
```

### Debug Output/Panic Output

```shell
Plan output : 
azurerm_resource_group.example: Refreshing state... [id=/subscriptions/xxxxxx/resourceGroups/example-resources]
azurerm_storage_account.example: Refreshing state... [id=/subscriptions/xxxxxx/resourceGroups/example-resources/providers/Microsoft.Storage/storageAccounts/terraformissuestorage01]

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_storage_account.example will be created
  + resource "azurerm_storage_account" "example" {
      + access_tier                        = (known after apply)
      + account_kind                       = "Storage"
      + account_replication_type           = "LRS"
      + account_tier                       = "Standard"
      + allow_nested_items_to_be_public    = true
      + cross_tenant_replication_enabled   = true
      + default_to_oauth_authentication    = false
      + dns_endpoint_type                  = "Standard"
      + enable_https_traffic_only          = true
      + id                                 = (known after apply)
      + infrastructure_encryption_enabled  = false
      + is_hns_enabled                     = false
      + large_file_share_enabled           = (known after apply)
      + local_user_enabled                 = true
      + location                           = "westeurope"
      + min_tls_version                    = "TLS1_2"
      + name                               = "terraformissuestorage01"
      + nfsv3_enabled                      = false
      + primary_access_key                 = (sensitive value)
      + primary_blob_connection_string     = (sensitive value)
      + primary_blob_endpoint              = (known after apply)
      + primary_blob_host                  = (known after apply)
      + primary_blob_internet_endpoint     = (known after apply)
      + primary_blob_internet_host         = (known after apply)
      + primary_blob_microsoft_endpoint    = (known after apply)
      + primary_blob_microsoft_host        = (known after apply)
      + primary_connection_string          = (sensitive value)
      + primary_dfs_endpoint               = (known after apply)
      + primary_dfs_host                   = (known after apply)
      + primary_dfs_internet_endpoint      = (known after apply)
      + primary_dfs_internet_host          = (known after apply)
      + primary_dfs_microsoft_endpoint     = (known after apply)
      + primary_dfs_microsoft_host         = (known after apply)
      + primary_file_endpoint              = (known after apply)
      + primary_file_host                  = (known after apply)
      + primary_file_internet_endpoint     = (known after apply)
      + primary_file_internet_host         = (known after apply)
      + primary_file_microsoft_endpoint    = (known after apply)
      + primary_file_microsoft_host        = (known after apply)
      + primary_location                   = (known after apply)
      + primary_queue_endpoint             = (known after apply)
      + primary_queue_host                 = (known after apply)
      + primary_queue_microsoft_endpoint   = (known after apply)
      + primary_queue_microsoft_host       = (known after apply)
      + primary_table_endpoint             = (known after apply)
      + primary_table_host                 = (known after apply)
      + primary_table_microsoft_endpoint   = (known after apply)
      + primary_table_microsoft_host       = (known after apply)
      + primary_web_endpoint               = (known after apply)
      + primary_web_host                   = (known after apply)
      + primary_web_internet_endpoint      = (known after apply)
      + primary_web_internet_host          = (known after apply)
      + primary_web_microsoft_endpoint     = (known after apply)
      + primary_web_microsoft_host         = (known after apply)
      + public_network_access_enabled      = true
      + queue_encryption_key_type          = "Service"
      + resource_group_name                = "example-resources"
      + secondary_access_key               = (sensitive value)
      + secondary_blob_connection_string   = (sensitive value)
      + secondary_blob_endpoint            = (known after apply)
      + secondary_blob_host                = (known after apply)
      + secondary_blob_internet_endpoint   = (known after apply)
      + secondary_blob_internet_host       = (known after apply)
      + secondary_blob_microsoft_endpoint  = (known after apply)
      + secondary_blob_microsoft_host      = (known after apply)
      + secondary_connection_string        = (sensitive value)
      + secondary_dfs_endpoint             = (known after apply)
      + secondary_dfs_host                 = (known after apply)
      + secondary_dfs_internet_endpoint    = (known after apply)
      + secondary_dfs_internet_host        = (known after apply)
      + secondary_dfs_microsoft_endpoint   = (known after apply)
      + secondary_dfs_microsoft_host       = (known after apply)
      + secondary_file_endpoint            = (known after apply)
      + secondary_file_host                = (known after apply)
      + secondary_file_internet_endpoint   = (known after apply)
      + secondary_file_internet_host       = (known after apply)
      + secondary_file_microsoft_endpoint  = (known after apply)
      + secondary_file_microsoft_host      = (known after apply)
      + secondary_location                 = (known after apply)
      + secondary_queue_endpoint           = (known after apply)
      + secondary_queue_host               = (known after apply)
      + secondary_queue_microsoft_endpoint = (known after apply)
      + secondary_queue_microsoft_host     = (known after apply)
      + secondary_table_endpoint           = (known after apply)
      + secondary_table_host               = (known after apply)
      + secondary_table_microsoft_endpoint = (known after apply)
      + secondary_table_microsoft_host     = (known after apply)
      + secondary_web_endpoint             = (known after apply)
      + secondary_web_host                 = (known after apply)
      + secondary_web_internet_endpoint    = (known after apply)
      + secondary_web_internet_host        = (known after apply)
      + secondary_web_microsoft_endpoint   = (known after apply)
      + secondary_web_microsoft_host       = (known after apply)
      + sftp_enabled                       = false
      + shared_access_key_enabled          = true
      + table_encryption_key_type          = "Service"
      + tags                               = {
          + "environment" = "staging"
        }

      + queue_properties {
          + logging {
              + delete                = true
              + read                  = true
              + retention_policy_days = 10
              + version               = "1"
              + write                 = true
            }
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

─────────────────────────────────────────────────────────────────────────────

Note: You didn't use the -out option to save this plan, so Terraform can't
guarantee to take exactly these actions if you run "terraform apply" now.

Apply output :

Error: `queue_properties` aren't supported for account kind "Storage" in sku tier "Standard"

  with azurerm_storage_account.example,
  on main.tf line 20, in resource "azurerm_storage_account" "example":
  20: resource "azurerm_storage_account" "example" {
```

### Expected Behaviour

According to the documentation, it should be possible to set queue_properties for Azure Storage Accounts with account_kind set to "Storage". Terraform should not throw an error when attempting to configure queue_properties for such accounts.

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#queue_properties

### Actual Behaviour

Terraform throws an error indicating that queue_properties are not supported for Azure Storage Accounts with account_kind set to "Storage"

### Steps to Reproduce

1. Set up a Terraform configuration file defining an Azure Storage Account with the following attributes:

> - account_kind set to "Storage"
> 
> - queue_properties defined with logging properties

2. Run terraform init and terraform apply with the provided configuration.
3. Observe the error message indicating that queue_properties are not supported for the specified account_kind.

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by magodo**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description

~Update document of `azurerm_storage_account` for the viable combinations for `(queue|share)_properties`~

`azurerm_storage_account` - Extend the support level of `(blob|queue|share)_properties` for `Storage` kind

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [ ] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

```shell
terraform-provider-azurerm on  storage_tier_kind_doc via 🐹 v1.22.1
💤  TF_ACC=1 go test -v -timeout=20h ./internal/services/storage -run='TestAccStorageAccount_StorageV1_'
=== RUN   TestAccStorageAccount_StorageV1_blobProperties
=== PAUSE TestAccStorageAccount_StorageV1_blobProperties
=== RUN   TestAccStorageAccount_StorageV1_queueProperties
=== PAUSE TestAccStorageAccount_StorageV1_queueProperties
=== RUN   TestAccStorageAccount_StorageV1_shareProperties
=== PAUSE TestAccStorageAccount_StorageV1_shareProperties
=== CONT  TestAccStorageAccount_StorageV1_blobProperties
=== CONT  TestAccStorageAccount_StorageV1_shareProperties
=== CONT  TestAccStorageAccount_StorageV1_queueProperties
--- PASS: TestAccStorageAccount_StorageV1_blobProperties (143.56s)
--- PASS: TestAccStorageAccount_StorageV1_queueProperties (158.50s)
--- PASS: TestAccStorageAccount_StorageV1_shareProperties (159.78s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/storage       159.808s
```

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_storage_account` - Extend the support level of `(blob|queue|share)_properties` for `Storage` kind [GH-00000]

This is a (please select all that apply):

- [x] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #25422

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: website/docs/r/storage_account.html.markdown [hunk 1]
```diff
@@ -136,14 +136,16 @@ The following arguments are supported:
 
 * `queue_properties` - (Optional) A `queue_properties` block as defined below.
 
-~> **NOTE:** `queue_properties` cannot be set when the `account_kind` is set to `BlobStorage`
+~> **NOTE:** `queue_properties` can only be set for `Standard` (`account_tier`) `StorageV2` (`account_kind`) Storage Accounts.
```
Comment (tombuildsstuff) [inline website/docs/r/storage_account.html.markdown:139]: Perhaps this'd be clearer as:

```suggestion
~> **NOTE:** `queue_properties` can only be configured when `account_tier` is set to `Standard` and `account_kind` is set to `StorageV2` Storage Accounts.
```

(assuming I've inferred that correctly from the message)?

File: website/docs/r/storage_account.html.markdown [hunk 2]
```diff
@@ -136,14 +136,16 @@ The following arguments are supported:
...
 
 * `static_website` - (Optional) A `static_website` block as defined below.
 
 ~> **NOTE:** `static_website` can only be set when the `account_kind` is set to `StorageV2` or `BlockBlobStorage`.
 
 * `share_properties` - (Optional) A `share_properties` block as defined below.
 
+~> **NOTE:** `share_properties` can only be set for `Standard` (`account_tier`) `StorageV2` (`account_kind`) or `Premium` (`account_tier`) `FileStorage` (`account_kind`) Storage Accounts.
```
Comment (tombuildsstuff) [inline website/docs/r/storage_account.html.markdown:147]: (as above) I think we could probably reword this to be clearer, are these two combinations (Standard+StorageV2 and Premium+FileStorage)? If so:

```suggestion
~> **NOTE:** `share_properties` can only be configured when either `account_tier` is `Standard` and `account_kind` is `StorageV2` - or when `account_tier` is `Premium` and `account_kind` is `FileStorage`.
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
Could we rephrase these to be clearer?

**Comment by rcskosir** [comment]
@magodo :wave: Thank you for using our new PR template! Can you please update the template to include the description for this PR under the description heading? This helps with the review process as well as traceability in the future for our users.

File: website/docs/r/storage_account.html.markdown [hunk 3]
```diff
@@ -136,14 +136,16 @@ The following arguments are supported:
 
 * `queue_properties` - (Optional) A `queue_properties` block as defined below.
 
-~> **NOTE:** `queue_properties` cannot be set when the `account_kind` is set to `BlobStorage`
+~> **NOTE:** `queue_properties` can only be configured when `account_tier` is set to `Standard` and `account_kind` is set to `StorageV2` Storage Accounts.
```
Comment (tombuildsstuff) [inline website/docs/r/storage_account.html.markdown:139]:
```suggestion
~> **NOTE:** `queue_properties` can only be configured when `account_tier` is set to `Standard` and `account_kind` is set to `StorageV2`.
```

we can ditch `Storage Accounts` here?

**Review by tombuildsstuff — APPROVED** [review]
Otherwise LGTM 👍

**Comment by MedFazazi** [comment]
Hi there, have you had a chance to check out my comment on this issue?

https://github.com/hashicorp/terraform-provider-azurerm/issues/25422#issuecomment-2024173406

**Comment by magodo** [comment]
@tombuildsstuff As @MedFazazi has mentioned, our existing feature check didn't account for the kind `Storage`, I've modified my PR to extend the support level of `(blob|queue|share)_properties` for `Storage` kind then. Please take another look!

**Comment by MedFazazi** [comment]
Hello @magodo,
Thank you for your efforts in extending the support level for blob_properties, queue_properties, and share_properties to the Storage kind. I wanted to kindly inquire if there's a possibility to also include support for configuring table_properties, including logging features, in Azure Storage Accounts. The ability to manage table service properties, including logging, through Terraform would greatly enhance its usability for various use cases.

Thank you for considering this request!

Warm regards,

File: website/docs/r/storage_account.html.markdown [hunk 4]
```diff
@@ -136,14 +136,16 @@ The following arguments are supported:
 
 * `queue_properties` - (Optional) A `queue_properties` block as defined below.
 
-~> **NOTE:** `queue_properties` cannot be set when the `account_kind` is set to `BlobStorage`
+~> **NOTE:** `queue_properties` can only be configured when `account_tier` is set to `Standard` and `account_kind` is set to `Storage(V2)`.
```
Comment (tombuildsstuff) [inline website/docs/r/storage_account.html.markdown:139]: The value for the constant is `StorageV2` and not `Storage(V2)`?

```suggestion
~> **NOTE:** `queue_properties` can only be configured when `account_tier` is set to `Standard` and `account_kind` is set to either `Storage` or `StorageV2`.
```

File: website/docs/r/storage_account.html.markdown [hunk 5]
```diff
@@ -136,14 +136,16 @@ The following arguments are supported:
...
 
 * `static_website` - (Optional) A `static_website` block as defined below.
 
 ~> **NOTE:** `static_website` can only be set when the `account_kind` is set to `StorageV2` or `BlockBlobStorage`.
 
 * `share_properties` - (Optional) A `share_properties` block as defined below.
 
+~> **NOTE:** `share_properties` can only be configured when either `account_tier` is `Standard` and `account_kind` is `Storage(V2)` - or when `account_tier` is `Premium` and `account_kind` is `FileStorage`.
```
Comment (tombuildsstuff) [inline website/docs/r/storage_account.html.markdown:147]:
```suggestion
~> **NOTE:** `share_properties` can only be configured when either `account_tier` is `Standard` and `account_kind` is either `Storage` or `StorageV2` - or when `account_tier` is `Premium` and `account_kind` is `FileStorage`.
```

**Review by tombuildsstuff — APPROVED** [review]
Left a couple of comments around the docs but otherwise 👍
